### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by notrexxx**
# Tic_Tac_Toe third Milestone

- Displaying prompts to users
- Win condition logic
- Input of character exceptions
- Styling

### CODE REVIEW

File: bin/main.rb [hunk 1]
```diff
@@ -1,16 +1,22 @@
 #!/usr/bin/env ruby
 require 'io/console'
+require 'colorize'
+require '../lib/board'
```
Comment (rukundoeric) [inline bin/main.rb]: Please use require_relative in this case, you can read more about why you should do that  [Here](https://medium.com/@ellishim/understanding-require-vs-require-relative-vs-require-all-80e3b26d89e6)

File: lib/board.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/AbcSize
```
Comment (rukundoeric) [inline lib/board.rb]: You should not overuse RuboCop disable cops comments, only complexity ones are allowed to be disabled.

**Review by rukundoeric — CHANGES_REQUESTED** [review]
# Status: ♻️ CHANGES REQUIRED ♻️
_Great job guys so far 👏 💯 👍. However, there are some changes required, please check the comments below._

- ### To Highlight
  - [x] The project is clean 💯
  - [x] It has a descriptive branch name 💯
  - [x] The project follows Git flow. 💯
  - [x] No `puts/print` and `gets.chomp` outside `main.rb` file.. 💯
- ### To Improve
  - [x] [MANDATORY] Please use require_relative in `main.rb on line 4`, you can read more about why you should do that Here;
  - [x] [MANDATORY] You should not overuse RuboCop disable cops comments, only complexity ones are allowed to be disabled.
  - [x] [MANDATORY] When I try to pray in one position twise, it is replacing the previous move when it should count it as invalid move, kindly fix that.


_Please make all the required changes before you submit another code review. If you have any problems or concerns, you can reach out to me on @slack: Rukundo Eric 🚀 . I wish you all the very best 🥇_

### _Keep working hard._
### _Happy Coding😃_
-----------------------------------------------------------------------------------------------------------------------------------------
**Regards,
[Rukundo Eric](https://rukundo-eric.herokuapp.com/) ,
[LinkedIn](https://www.linkedin.com/in/rukundo-eric/), 
@slack: Rukundo Eric**

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Changes Required :recycle: 
**HIGHLIGHTS:**
- Very nice work
- Good job implementing the changes from previous TSE

**Changes required**
- [x] When game ends as a draw your game never informs them that the game ended as draw, kindly fix this
- [ OPTIONAL ] The project uses at least two classes (Usually it uses three: Game, Board and Player). The code of the project should be logically divided between classes. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/classes.png)). I am adding this as optional because it was not asked by the first reviewer but I highly recommend you doing this.
- [ OPTIONAL] Use attr_readers/attr_writers instead of attr_accessor because by using attr_accessor you allow writing that property from outside the class which is not recommended

### **Please make the required changes and submit another review request when you are ready.**

<hr>
### if you have any question kindly reach me on slack @GzimAsani

### HAPPY CODING :computer:

**Review by smcommits — APPROVED** [review]
**STATUS: APPROVED**

Good job on implementing the changes requested. The project is now approved. You can go ahead and merge the branch. 

### Optional 
Your main.rb has 141 lines of loose code. Try to divide your code into smaller functions. Refer to [this](https://stackoverflow.com/questions/13256453/what-is-the-advantage-of-breaking-a-code-into-several-small-functions-in-c) stack-overflow answer to understand the importance of dividing your code. 

Again, excellent work! Be proud of yourself.

**Pull request merged**